### DISCUSSION TITLE
Expecting /token endpoint in authorization code flow to return scope?

### DISCUSSION THREAD

**Discussion by rustyb** [zitadel/zitadel#6594] [Q&A]
👋 Hi Folks - Hoping you can help me with a little issue i'm having here. 

Reading the docs for the token endpoint https://zitadel.com/docs/apis/openidoauth/endpoints#token_endpoint
<img width="773" alt="Screenshot 2023-09-19 at 09 32 17" src="https://github.com/zitadel/zitadel/assets/53086/49cdf00c-0bb8-493b-96a7-9e5d27a12f54">

I was expecting there to be a `scope` value in the return json. But I receive:

```json
{
  "access_token": "<JWT>",
  "token_type": "Bearer",
  "expires_in": 43199,
  "id_token": "<JWT>"
}
```

To reproduce:
1. Created a project
2. Created a web app - downloaded the client_id and secret
3. Make a call to authorize
```
http://localhost:8080/oauth/v2/authorize?
client_id=client_id@test
&redirect_uri=http://localhost/auth/zitadel/callback
&response_type=code
&scope=openid+email+profile+urn:zitadel:iam:org:project:roles+urn:zitadel:iam:user:metadata
&code_challenge=something-code-challenge
&code_challenge_method=S256
```
4. With the returned code request a token:
```bash
curl --request POST \
  --url http://localhost:8080/oauth/v2/token \
  --header 'Content-Type: multipart/form-data' \
  --form code=SYA7mJ0FPC75QI0doSzOBVqk3nnyK4aDXwEPfxfRWhxfTg \
  --form grant_type=authorization_code \
  --form redirect_uri=http://localhost/auth/zitadel/callback \
  --form code_verifier=something-code-challenge \
  --form client_id=clilent_id@test \
  --form client_secret=<CLIENT_SECRET>
```
5. Token endpoint returns:
```json
{
  "access_token": "<JWT>",
  "token_type": "Bearer",
  "expires_in": 43199,
  "id_token": "<JWT>"
}
```

Is there some other parameter I need to request?

I was expecting to get a scope value like you can expect from google oidc flow:
```
{
	"access_token": "__MOCK_ACCESS_TOKEN__",
	"expires_in": 3599,
	"scope": "openid https://www.googleapis.com/auth/userinfo.email",
	"token_type": "Bearer",
	"id_token": "__MOCK_ID_TOKEN__",
}
```

Any ideas?

**Comment by hifabienne** [comment]
Hei @rustyb 
We just had a look at the specification.
It is optional to return them, when the requested scope is the same.
Otherwise it should be returned.
I will turn the discussion into a issue.

**Reply by rustyb** [reply]
Ok thanks for having a look!

**Reply by mgius-acurade** [reply]
Hi rustyb, any updates?

**Reply by rustyb** [reply]
Sorry i had to abandon the project I was working on so no updates unfortunately.
